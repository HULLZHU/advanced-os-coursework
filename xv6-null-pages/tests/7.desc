testing mprotect and munprotect by using it on the page that main() is mapped to and attempting to write things to main.

When you run the test_7.c in xv6, the program will crash & you should be getting an error message like:
"pid 8 test_7: trap 14 err 7 on cpu 1 eip 0x1020 addr 0x1000--kill proc"