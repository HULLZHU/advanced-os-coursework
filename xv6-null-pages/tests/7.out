XV6_TEST_OUTPUT: Return value of mprotect on Page where MAIN is mapped: 0
